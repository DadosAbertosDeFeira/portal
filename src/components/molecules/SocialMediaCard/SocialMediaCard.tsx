import type { SocialMediaType } from "@utils/social";
import { handleSocialMediaLink, lookupSocialMediaIcon } from "@utils/social";
import { Box, type BoxProps } from "atoms/Box";
import { Image } from "atoms/Image";
import { Link } from "atoms/Link";
import React from "react";

export type SocialMediaCardProps = BoxProps & {
  type: SocialMediaType;
  user: string;
};

export function SocialMediaCard({
  type,
  user,
  ...props
}: SocialMediaCardProps) {
  return (
    <Box className="mr-4" {...props}>
      <Link
        href={handleSocialMediaLink(type, user)}
        target="_blank"
        rel="noreferrer"
      >
        <Image
          width={20}
          height={20}
          src={lookupSocialMediaIcon(type)}
          alt={`${type} icon`}
        />
      </Link>
    </Box>
  );
}
