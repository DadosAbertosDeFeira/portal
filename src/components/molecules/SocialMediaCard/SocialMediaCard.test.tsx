import { render } from "@testing-library/react";

import { SocialMediaCard, type SocialMediaCardProps } from "./SocialMediaCard";

describe("SocialMediaCard", () => {
  const makeSut = (props: SocialMediaCardProps) => {
    return render(<SocialMediaCard {...props} />);
  };

  it("should render github card correctly", () => {
    const { asFragment } = makeSut({ type: "github", user: "teste" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render instagram card correctly", () => {
    const { asFragment } = makeSut({ type: "instagram", user: "teste" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render linkedin card correctly", () => {
    const { asFragment } = makeSut({ type: "linkedin", user: "teste" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render twitter card correctly", () => {
    const { asFragment } = makeSut({ type: "twitter", user: "teste" });

    expect(asFragment).toMatchSnapshot();
  });
});
