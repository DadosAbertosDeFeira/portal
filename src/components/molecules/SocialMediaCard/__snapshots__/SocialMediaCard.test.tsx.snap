// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaCard should render github card correctly 1`] = `[Function]`;

exports[`SocialMediaCard should render instagram card correctly 1`] = `[Function]`;

exports[`SocialMediaCard should render linkedin card correctly 1`] = `[Function]`;

exports[`SocialMediaCard should render twitter card correctly 1`] = `[Function]`;
